ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/LdapService.java

```diff
@@ -110,73 +52,21 @@ private AuthenticationResult authenticateUser(String userInfoName, String passwo
 
         try {
             DirContext userContext = new InitialDirContext(userLdapProperties);
-            userAuthenticated = true;
             userContext.close(); // close can also throw an exception
         } catch (NamingException e) {
             if (e instanceof AuthenticationException) {
-                accountLocked = updateUserFailedLoginAttempts(userInfoName);
```

Comment:
> I split LdapService into two classes: LdapService and UserService. I didn't rewrite the functionality, but just moved the method around.

This is the one exception. `authenticateUser` is an LDAP method for certain, but it calls `updateUserFailedLoginAttempts` in its implementation. This creates a circular dependency between `UserService` and `LdapService` which is hard to understand and test. I moved `updateUserFailedLoginAttempts` out of this method and into `UserService.authenticate`.